### PR TITLE
tvOS

### PR DESCRIPTION
**Pull request by djih**
Adds support for tvOS
   * uses the cache directory for storage if on tvOS instead of the library directory
   * change constant values for platform and os
   * set `kAmpEventUploadThreshold` to 1 for tvOS
   * update the 413 backoff logic to handle when event batchsize is already 1
   * update the podspec
   * update to OCMock 3.2.1 which supports tvOS (unfortunately does not fix flakey test)
   * fix tests that broke with the new OCMock version
   * add a new target in the Xcode project which runs the unit tests in tvOS simulator
   * separate out iOS-specific and tvOS-specific tests

Verified that logging events works by instrumenting a demo tvOS app: https://github.com/amplitude/tvOS-Demo-App

### CODE REVIEW

File: Amplitude/AMPUtils.m [hunk 1]
```diff
@@ -141,4 +141,13 @@ + (NSDictionary *) validateGroups:(NSDictionary *) obj
     return [NSDictionary dictionaryWithDictionary:dict];
 }
 
++ (NSString*) platformDataDirectory
+{
+#if TARGET_OS_TV
+    return [NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) objectAtIndex: 0];
```
Comment (curtisliu) [inline Amplitude/AMPUtils.m]: why do we use NSCaches for tvOS?

**Comment by gabek** [comment]
Just tested the tvOS branch in my own application and it functions as expected.

File: Amplitude/AMPUtils.m (hunk 1 above)
Comment (gabek) [inline Amplitude/AMPUtils.m]: The cache directory is valid in tvOS, but the data stored there is only guaranteed to be there while the application is running.

Comment (djih) [inline Amplitude/AMPUtils.m]: Unfortunately it seems there's no persistent storage for tvOS apps, so we need to upload events immediately

**Pull request merged**